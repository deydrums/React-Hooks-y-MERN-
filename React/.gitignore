# proyectos
/00-proyectos