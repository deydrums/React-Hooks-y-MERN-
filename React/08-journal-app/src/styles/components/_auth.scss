.auth__main {
    align-items: center;
    //background-image: linear-gradient(to top, rgba(228, 149, 248, 0.986) 0%, rgba(52, 15, 73, 0.4) 100%), url(../../assets/auth/portada.jpg);
    background-image: linear-gradient(to top, rgba(59, 19, 104, 0.986) 0%, rgba(67, 43, 151, 0.829) 100%);
    display: flex;
    justify-content: center;
    margin:0px;
    height: 100vh;
    width: 100vw;
}

.auth__box-container{
    border-radius: 10px;
    background-color: rgba(255, 255, 255, 0.733);
    box-shadow: 0px 0px 25px rgba(0, 0, 0, 0.9);
    padding: 20px;
    width: 270px;
    transition: all 0.5s linear;
}

.auth__box-container:hover{
    transform: scale(1.01);
}

.auth__title{
    color: darken($color: $primary, $amount: 20);
    margin-bottom: 15px;
    font-size:22px;
}

.auth__input{
    color: $dark-grey;
    border: 0px;
    border-bottom: 1px solid $light-grey;
    font-size: 16px;
    margin-bottom:10px;
    height: 25px;
    width: 100%;
    transition: border-bottom .3s ease;
    border-radius: 3px;

    &:focus{
        border-bottom: 1px solid $primary;
        outline:none;
    }
}

.auth__social-networks{
    align-items: center;
    display: flex;
    justify-content: center;
    flex-direction: column;
    padding-top: 20px;
    padding-bottom: 20px;
    width: 100%;
}

.auth__alert-error{
    background-color: rgb(190, 95, 95);
    border-radius: 5px;
    color: #fff;
    display: flex;
    justify-content: center;
    margin-bottom: 10px;
    padding: 5px;
}

/*Page loader*/
.page-loader{
    position:fixed;
    left:0;
    top:0;
    height:100%;
    width:100%;
    z-index: 999;
    background-color: var(--white-alpha-25);
    backdrop-filter: var(--backdrop-filter-blur);
    display: flex;
    align-items: center;
    justify-content:center;
    transition: opacity 0.6s ease;

}
.page-loader.fade-out{
    opacity: 0;
}

.page-loader div{
    border: 2px solid transparent;
    position:absolute;
    border-radius: 50%;
    opacity: 0.6;
    border-top-color: transparent !important;
    border-bottom-color: transparent !important;
    animation: spin 1s linear infinite;
}

.page-loader div:nth-child(1){
    height:60px;
    width:60px;
    border-color: $light-blue;
    animation-duration: 1.2s;
}

.page-loader div:nth-child(2){
    height:45px;
    width:45px;
    border-color: $green;
}

.page-loader div:nth-child(3){
    height:30px;
    width:30px;
    border-color: $orange;
}


@keyframes spin {
    0%{
        transform: rotate(0deg);
    }
    100%{
        transform: rotate(360deg);
    }
}


@media only screen and (max-width: 600px) {
    .auth__box-container{
        border-radius: 20px;
        width: 80vw;
        height: 400px;
    }
    .auth__input{
        margin-bottom:15px;
        height: 35px;
    }
  }