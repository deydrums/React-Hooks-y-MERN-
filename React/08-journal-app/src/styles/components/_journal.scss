.journal__main-content{
    display: flex;
    width: 100vw;
    overflow: hidden;
}

.journal__sidebar {
    background-color: $dark-grey;
    color: white;
    display: flex;
    flex-direction: column;
    height: 100vh;
    padding-left: 10px;
    padding-right: 10px;
    width: 350px;
    transition: transform .3s ease-in-out;
    transform: translate(0%);
}

.journal__sidebar-navbar{
    display: flex;
}

.journal__sidebar-btnlogout{
    width: 100%;
    text-align:right;
    margin-bottom: 10px;
}

.journal__sidebar-navbar i{
    margin-top: 6px;
    color: $light-grey;
}

.journal__sidebar-navbar h3{
    font-weight: lighter;
    margin-left: 7px !important;
    color: $light-blue;
}

.journal__new-entry{
    align-items: center;
    cursor: pointer !important;
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-top: 30px;
    width: 100%;
    transition: color .3s ease;
    &:hover{
        color: darken($color: #fff, $amount: 20)
    }
}

.journal__entries {
    margin-top: 30px;
    overflow: auto;
 
    &::-webkit-scrollbar {
        -webkit-appearance: none;
        background-color: #ccc;
    }
 
    &::-webkit-scrollbar:vertical {
        width: 8px;
    }
 
    &::-webkit-scrollbar:horizontal {
        height: 10px;
    }
 
    &::-webkit-scrollbar-thumb {
        background-color: $primary;
        border-radius: 20px;
    }
}

.journal__entry{
    background-color:#fff;
    border-radius: 5px;
    color: $dark-grey;
    display: flex;
    margin-bottom: 10px;
    overflow: hidden;
}

.journal__entry-picture{
    height: 75px;
    width: 75px;
}


.journal__entry-body {
    padding: 10px;
}

.journal__entry-title {
    font-size: 14px;
    font-weight: bold;
}

.journal__entry-content {
    font-size: 10px;
}

.journal__entry-date-box {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 5px;
}

.journal__entry-date-box span{
    font-size: 12px;
}

.journal__btnmore{

    padding: 15px;
    background-color:$primary;
    border-radius: 50%;
    position: absolute;
    cursor: pointer;
    bottom:10px;
    right:10px;
    animation: zoomInOut 8s linear infinite;
    color: darken($color: #fff, $amount: 10);
    &:hover{
        outline: none !important;
    }
    display:none;
}

main{
    height:100vh;
}

@keyframes zoomInOut {
    0%,100%{
        transform: scale(0.8);
    }
    50%{
        transform: scale(1);
    }
}


@media only screen and (max-width: 600px) {

    .journal__sidebar {
        position:fixed;
        width: 300px;
        transform: translate(-100%);
        transition: transform .3s ease-in-out;
    }
    .journal__active{
        transform: translate(0);
    }
    .journal__btnmore{
        display:block;
    }
}