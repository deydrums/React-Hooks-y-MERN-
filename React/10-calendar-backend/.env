PORT=80
DB_CNN=mongodb+srv://deydrums:7VaOymNqlPNdUn12@cluster0.id2bg.mongodb.net/CalendarApp
SECRET_JWT_SEED=Esto-Es-una-PalAbRa-S3kr2t4-j3j3j3