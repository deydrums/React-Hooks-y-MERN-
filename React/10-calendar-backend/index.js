const express = require('express');
require('dotenv').config();
//Ejecutar express
var app = express();

//Directorio publico
app.use(express.static('public'));

//Rutas
//Reescribir rutas
app.use('/api/auth', require('./routes/auth'));

//TODO: CRUD: Eventos

//Escuchar peticiones
app.listen(process.env.PORT,()=>{
    console.log(`Servidor corriendo en puerto ${process.env.PORT}`)
});