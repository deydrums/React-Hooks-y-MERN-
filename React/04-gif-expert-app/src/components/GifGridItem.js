import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({title,url}) => {
    return (
            <a className="imagen-port animate__animated animate__fadeIn" href={url} target="_blank">
                <img className = 'gif' src={url} alt={title} />
                <div className="hover-galeria">
                    <img src ="/images/icono1.png" />
                    <p>Abrir</p>
                </div>
            </a>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}



/*
    1.Enzyme
    2.Enzyme to Json
    3.Debe de mostrar el componente correctamente
        *shallow
        *wrapper
        *wraper .toMatchSnapshot()

*/