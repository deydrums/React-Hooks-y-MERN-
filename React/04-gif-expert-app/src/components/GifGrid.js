import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {
    
    const {data:images, loading} = useFetchGifs(category);
    return (
        <>
        <hr></hr>
            <section className="portafolio">
                <div className="contenedor">
                    <h2 className="titulo animate__animated animate__fadeIn">{category}</h2>
                    <div className="galeria-port">
                    {loading && <p className ="animate__animated animate__flash">Loading</p>}

                        
                        {
                            images.map(img =>(
                                <GifGridItem 
                                    key = {img.id}
                                    {...img}
                                />
                            ))
                        }
                    </div>
                    </div>
                </section>
        </>
    )
}

GifGrid.prototype = {
    category: PropTypes.string.isRequired
}
