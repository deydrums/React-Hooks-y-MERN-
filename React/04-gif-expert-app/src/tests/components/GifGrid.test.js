import { GifGrid } from "../../components/GifGrid"
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import React from 'react';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid></GifGrid>', () => {
    const category = 'Dragon Ball';

    test('Debe de mostrarse el componente <GifGrid></GifGrid>', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category = {category}/>);
        expect (wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        expect (wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
})
