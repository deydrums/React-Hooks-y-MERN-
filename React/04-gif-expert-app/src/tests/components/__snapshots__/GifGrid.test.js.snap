// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid></GifGrid> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <hr />
  <section
    className="portafolio"
  >
    <div
      className="contenedor"
    >
      <h2
        className="titulo animate__animated animate__fadeIn"
      >
        Dragon Ball
      </h2>
      <div
        className="galeria-port"
      >
        <GifGridItem
          id="ABC"
          key="ABC"
          title="ABC Title"
          url="https://google.com"
        />
      </div>
    </div>
  </section>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid></GifGrid> Debe de mostrarse el componente <GifGrid></GifGrid> 1`] = `
<Fragment>
  <hr />
  <section
    className="portafolio"
  >
    <div
      className="contenedor"
    >
      <h2
        className="titulo animate__animated animate__fadeIn"
      >
        Dragon Ball
      </h2>
      <div
        className="galeria-port"
      >
        <p
          className="animate__animated animate__flash"
        >
          Loading
        </p>
      </div>
    </div>
  </section>
</Fragment>
`;
